bindings/python/example_scripts/export_account_totals.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+# export_account_totals -- export the account totals of all accounts.
+
+# Copyright (C) 2020 Hong Xu <hong@topbug.net>
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# @author Hong Xu <hong@topbug.net>
+
+# Usage:
+#
+#    python3 export_account_totals.py /path/to/gnucash_book_file > output.csv
+
+import sys
+from decimal import Decimal
+
+import gnucash
+from gnucash import Session, SessionOpenMode
+
+
+def get_all_sub_accounts(account, names=[]):
+    "Iterate over all sub accounts of a given account."
+
+    for child in account.get_children_sorted():
+        child_names = names.copy()
+        child_names.append(child.GetName())
+        yield child, '::'.join(child_names)
+        yield from get_all_sub_accounts(child, child_names)
+
+
+if __name__ == '__main__':
+    print('Name,Commodity,Totals,Totals (USD)')
+    with gnucash.Session(sys.argv[1], SessionOpenMode.SESSION_READ_ONLY) as session:
+        commodity_table = session.book.get_table()
+        USD = commodity_table.lookup("ISO4217", "USD")
+        root_account = session.book.get_root_account()
+        for account, account_name in get_all_sub_accounts(root_account):
+            print(','.join([str(account_name),
+                            account.GetCommodity().get_mnemonic(),
+                            str(round(float(account.GetBalance().to_fraction()), 2)),
+                            str(round(float(account.GetBalanceInCurrency(USD, True).to_fraction()), 2))
```

Comment:
Don't ever use floating point in financial software. Use `gnc_numeric_to_string`.

Reply:
If I use `gnc_numeric_to_string`, it outputs things like `12345/100`, which I don't think is what most people want. Do you know whether there's a helper that converts a gnc_numeric to a decimal-looking string?

Reply:
I added my own helper to the script. Feel free to let me know if there is already a more robust helper within gnucash!